docs/data/data-grid/recipes-row-grouping/RowGroupingCheckboxSelection.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  GRID_CHECKBOX_SELECTION_COL_DEF,
+  GridCellCheckboxRenderer,
+  GridColDef,
+  GridColumnHeaderParams,
+  gridFilteredSortedRowIdsSelector,
+  GridHeaderCheckbox,
+  GridRenderCellParams,
+  selectedIdsLookupSelector,
+  useGridApiRef,
+  useKeepGroupedColumnsHidden,
+  GridApi,
+} from '@mui/x-data-grid-premium';
+import { useMovieData } from '@mui/x-data-grid-generator';
+
+const checkboxColumn = (apiRef: React.MutableRefObject<GridApi>): GridColDef => {
+  return {
+    ...GRID_CHECKBOX_SELECTION_COL_DEF,
+    renderHeader: (params) => {
+      const children = gridFilteredSortedRowIdsSelector(
+        apiRef.current.state,
+        apiRef.current.instanceId,
+      ).filter((id) => !id.toString().includes('auto-generated'));
+
+      const selectionLookup = selectedIdsLookupSelector(
```

Comment:
Could we use `useGridSelector` here and do not recompute all of this on every render?
For large dataset we seem to have a potential bottleneck.

Reply:
I just noticed that we do have a PR open for the parent/chilren selection feature: #13757 ... This would probably make this recipe obsolete ... 🤷🏼